### ISSUE TITLE
patching multiple properties on a jsonb field only sets last property

### ISSUE THREAD

**Issue by thejuan** [Vincit/objection.js#865]
`Entity
      .query()
      .patch({"data:field1":1, "data:field2":2})
      .where("id", 5)
      .returning("*")`

Only one of the fields on the jsonb is updated.
I would expect it to have translated to

`set data = data || jsonb '{field1:1, field2:2}'`

Is there anyway to tell objection.js to use concat?

**Comment by koskimas** [comment]
How would that work with deeply nested fields like:

```js
Entity.query() .patch({"data:field1.field2.field3":1, "data:field2":2}) .where("id", 5) .returning("*")
```

**Comment by koskimas** [comment]
We could probaly implement this using nested jsonb_set calls:

```js
jsonb_set(jsonb_set("data", '{field1}, ?, true), '{field2}', ?, true))
```

**Comment by koskimas** [comment]
@thejuan This is now fixed in 1.1.1. Just run `npm install`.

**Comment by yisibl** [comment]
Starting with PostgreSQL v14, we can use DX's better `jsonb Subscripting`.

https://www.postgresql.org/docs/17/datatype-json.html#JSONB-SUBSCRIPTING
